crates/js-sys/src/lib.rs

```diff
@@ -3561,6 +3561,33 @@ pub mod WebAssembly {
         pub fn set(this: &Table, index: u32, function: &Function) -> Result<(), JsValue>;
     }
 
+    
+    // WebAssembly.Global
+    #[wasm_bindgen]
+    extern "C" {
+        /// The `WebAssembly.Global()` constructor creates a new `Global` object
+        /// of the given type and value.
+        ///
+        /// [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/Global)
+        #[wasm_bindgen(js_namespace = WebAssembly, extends = Object, typescript_type = "WebAssembly.Global")]
+        #[derive(Clone, Debug, PartialEq, Eq)]
+        pub type Global;
+
+        /// The `WebAssembly.Global()` constructor creates a new `Global` object
+        /// of the given type and value.
+        ///
+        /// [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/Global)
+        #[wasm_bindgen(constructor, js_namespace = WebAssembly, catch)]
+        pub fn new(global_descriptor: &Object, value: Option<usize>) -> Result<Global, JsValue>;
```

Comment:
I'm unsure if here value should be `Option<f64>` instead

Reply:
I think `Object` here should probably be a `GlobalDescriptor` dictionary with typed field accesses, and `value` should be `JsValue` most likely to support `externref` globals.

Reply:
Done. I set `value` to `Option<&JsValue>`.

For the `global_descriptor` I used the same pattern as `WebAssembly.Table` or `WebAssembly.Module`: using `Object`.

I do agree that having a `GlobalDescriptor` dictionary would be safer, but I'm not clear on how to do it looking at the code. Any tips here would be appreciated.